src/pages/manager/OpenAccountPage.js

```
@@ -1,13 +1,38 @@
 import { expect } from '@playwright/test';
 
-export class AddCustomerPage {
+export class OpenAccountPage {
   constructor(page) {
     this.page = page;
+    this.selectCustomerLocator = page.locator('select[name="userSelect"]');
+    this.selectCurrencyLocator = page.locator('select[name="currency"]');
   }
 
   async open() {
-    await this.page.goto(
-      '/angularJs-protractor/BankingProject/#/manager/openAccount',
-    );
+    await this.page.goto('#/manager/openAccount');
+  }
+
+  async selectCustomerOption(fullName) {
+    await this.selectCustomerLocator.selectOption({ label: fullName });
+  };
+
+  async selectCurrencyOption(currency) {
+    await this.selectCurrencyLocator.selectOption({ value: currency });
+  }
+
+  async selectRandomCurrency() {
+    const options = this.selectCurrencyLocator.locator('option');
+    const count = await options.count();
+    const randomIndex = Math.floor(Math.random() * (count - 1)) + 1;
+    const value = await options.nth(randomIndex).getAttribute('value');
```

selectRandomCurrency: `const value = await options.nth(randomIndex).getAttribute('value');` can return null if the attribute is missing. Validate `value` before calling `selectOption(value)` and handle the null case (fallback to selecting by index or by visible text). Otherwise `selectOption(null)` may throw or silently fail.